traceon/geometry.py

```diff
@@ -673,6 +670,126 @@ def aperture(height, radius, extent, z=0.):
+        plane_normal = np.array(plane_normal, dtype=float)
+        direction = np.array(direction, dtype=float)
+
+        dot_product = np.dot(direction, plane_normal)
+
+        if not np.isclose(dot_product, 0, atol=1e-7):
+            corrected_direction = direction - (dot_product / np.linalg.norm(plane_normal)**2) * plane_normal
+            raise AssertionError(
+                f"The provided direction {direction} does not lie in the specified plane. \n"
+                f"The closed valid direction is {np.round(corrected_direction, 10)}.")
+        
+        direction /= np.linalg.norm(direction)
+        plane_normal /= np.linalg.norm(plane_normal)
+
+        if angle < 0:
+            direction, angle = -direction, -angle
+
+        center = start - radius * np.cross(direction, plane_normal)
+        center_to_start = start - center
+        
+        def f(l):
+            theta = l/radius
+            return center + np.cos(theta) * center_to_start + np.sin(theta)*np.cross(plane_normal, center_to_start)
+        
+        return Path(f, radius*angle)
+    
+    def arc_to_polar(self, radius, angle, plane_normal=[0, 1, 0]):
+        """Extend the current path by a smooth arc using polar coordinates.
+        The arc is defined by a specified radius and angle and rotates counterclockwise
+         around around the normal that defines the arcing plane.
+
+        Parameters
+        ---------------------------
+        radius : float
+            The radius of the arc
+        angle : float
+            The angle subtended by the arc (in radians)
+        plane_normal : (3,) float
+            The normal vector of the plane containing the arc
+
+        Returns
+        ----------------------------
+        Path"""
+        plane_normal = np.array(plane_normal, dtype=float)
+        start_point = self.endpoint()
+        tangent = self.tangent(self.path_length)
+
+        dot_product = np.dot(tangent, plane_normal)
+        if not np.isclose(dot_product, 0,atol=1e-7):
+
+            corrected_normal = plane_normal - (dot_product / np.linalg.norm(tangent)**2) * tangent
+            raise AssertionError(
+                f"The provided plane normal {plane_normal} is not orthogonal to the tangent {tangent}  \n"
+                f"of the path at the endpoint so no smooth arc can be made. The closest valid normal is "
+                f"{np.round(corrected_normal, 10)}.")
+        
+        plane_normal /= np.linalg.norm(plane_normal)
+
+        return self >> Path.arc_polar(radius, angle, start_point, plane_normal, tangent)
+
+    def reversed(self):
+        """Generate a reversed version of the current path.
+        The reversed path is created by inverting the traversal direction,
+        such that the start becomes the end and vice versa.
+
+        Returns
+        ----------------------------
+        Path"""
+        return Path(lambda t: self(self.path_length-t), self.path_length, self.breakpoints, self.name)
```

Comment:
You forgot to invert the breakpoints too:
`breakpoints = [self.path_length - b for b in self.breakpoints]`